lib/tree_stand/node.rb

```diff
@@ -164,6 +164,66 @@ def each(&block)
       enumerator
     end
 
+    # Enumerate named children.
+    # @example
+    #   node.text # => "3 * 4"
+    #
+    # @example Iterate over the child nodes
+    #   node.each do |child|
+    #     print child.text
+    #   end
+    #   # prints: 34
+    #
+    # @example Enumerable methods
+    #   node.map(&:text) # => ["3", "4"]
+    #
+    # @yieldparam child [TreeStand::Node]
+    sig do
+      params(block: T.nilable(T.proc.params(node: TreeStand::Node).returns(BasicObject)))
+        .returns(T::Enumerator[TreeStand::Node])
+    end
+    def each_named(&block)
+      enumerator = Enumerator.new do |yielder|
+        @ts_node.each_named do |child|
+          yielder << TreeStand::Node.new(@tree, child)
+        end
+      end
+      enumerator.each(&block) if block_given?
+      enumerator
+    end
```

Comment:
Is there any advantage in creating directly an Enumerator rather than using `enum_for`?  I find the latter more natural.

For instance (untested)
```ruby
def each_named(&block)
  return enum_for __method__ if !block_given?

  @ts_node.each_named do |child|
    yield TreeStand::Node.new(@tree, child)
  end
end
```

Reply:
In terms of types, this returns a `Range[Integer]`, not an `Enumerator[Node]`.